src/transcodeImage.js

```diff
@@ -0,0 +1,283 @@
+const dicomCodec = require("@cornerstonejs/dicom-codec");
+const Tags = require("./Tags");
+
+const transcodeOp = {
+  none: 0,
+  decodeOnly: 1,
+  encodeOnly: 2,
+  transcode: 3, // decode and then encode
+};
+
+/**
+ * Static mapping for transcoding image.
+ */
+const transcodeMap = {
+  "1.2.840.10008.1.2": {
+    transferSyntaxUid: "1.2.840.10008.1.2",
+    transcodeOp: transcodeOp.decodeOnly,
+    alias: "uncompressed",
+  },
+  "1.2.840.10008.1.2.1": {
+    transferSyntaxUid: "1.2.840.10008.1.2.1",
+    transcodeOp: transcodeOp.decodeOnly,
+    alias: "uncompressed",
+  },
+  "1.2.840.10008.1.2.2": {
+    transferSyntaxUid: "1.2.840.10008.1.2.1",
+    transcodeOp: transcodeOp.decodeOnly,
+    alias: "uncompressed",
+  },
+  "1.2.840.10008.1.2.4.57": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jpeglossless",
+  },
+  "1.2.840.10008.1.2.4.70": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jpeglossless",
+  },
+  "1.2.840.10008.1.2.4.90": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jp2",
+  },
+  "1.2.840.10008.1.2.4.91": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jp2",
+  },
+  "1.2.840.10008.1.2.5": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "rle",
+  },
+};
+
+/**
+ * Minimum image info data to be used on transcode process by dicom-codec api.
+ */
+function getImageInfo(dataSet) {
```

Comment:
dicom-codec will adapt this set to each codec and this is the minimum set for it.